Assets/Scripts/Gameplay/UI/NetworkSimulationUIMediator.cs

```diff
@@ -0,0 +1,240 @@
+            m_ScenariosDropdown.onValueChanged.AddListener(OnScenarioChanged);
+
+            // Hide UI until ready
+            Hide();
+        }
+
+        void Start()
+        {
+            NetworkManager.Singleton.OnClientStarted += OnNetworkManagerStarted;
+            NetworkManager.Singleton.OnServerStarted += OnNetworkManagerStarted;
+        }
+
+        void OnDestroy()
+        {
+            if (NetworkManager.Singleton is not null)
+            {
+                NetworkManager.Singleton.OnClientStarted -= OnNetworkManagerStarted;
+                NetworkManager.Singleton.OnServerStarted -= OnNetworkManagerStarted;
+            }
+        }
+
+        void OnNetworkManagerStarted()
+        {
+            if (m_NetworkSimulator.IsAvailable)
+            {
+                Show();
+            }
+        }
+
+        void OnPresetChanged(int optionIndex)
+        {
+            m_NetworkSimulator.ChangeConnectionPreset(m_SimulatorPresets[m_PresetsDropdown.options[optionIndex].text]);
+        }
+
+        void OnScenarioChanged(int optionIndex)
+        {
+            var scenarioName = m_ScenariosDropdown.options[optionIndex].text;
+            m_NetworkSimulator.Scenario = m_Scenarios.ContainsKey(scenarioName) ? m_Scenarios[scenarioName] : null;
+            m_NetworkSimulator.Scenario?.Start(m_NetworkSimulator);
+            UpdateScenarioButton();
+        }
+
+        public void Hide()
+        {
+            m_CanvasGroup.alpha = 0f;
+            m_CanvasGroup.interactable = false;
+            m_CanvasGroup.blocksRaycasts = false;
+            m_Shown = false;
+        }
+
+        void Show()
+        {
+            m_CanvasGroup.alpha = 1f;
+            m_CanvasGroup.interactable = true;
+            m_CanvasGroup.blocksRaycasts = true;
+            UpdateScenarioButton();
+            m_Shown = true;
+        }
+
+        void ToggleVisibility()
+        {
+            if (m_Shown)
+            {
+                Hide();
+            }
+            else
+            {
+                Show();
+            }
+        }
+
+        void Update()
+        {
+            if (m_NetworkSimulator.IsAvailable)
+            {
+                if (Input.touchCount == k_NbTouchesToOpenWindow && AnyTouchDown() ||
+                    m_OpenWindowKeyCode != KeyCode.None && Input.GetKeyDown(m_OpenWindowKeyCode))
+                {
+                    ToggleVisibility();
+                }
+
+                var selectedPreset = m_PresetsDropdown.options[m_PresetsDropdown.value].text;
+                if (selectedPreset != m_NetworkSimulator.CurrentPreset.Name)
+                {
+                    for (var i = 0; i < m_PresetsDropdown.options.Count; i++)
+                    {
+                        if (m_PresetsDropdown.options[i].text == m_NetworkSimulator.CurrentPreset.Name)
+                        {
+                            m_PresetsDropdown.value = i;
+                        }
+                    }
+                }
```

Comment:
This bit of code is here for when we have the ConnectionsCycle scenario running right? If so, is there a way this bit of code can be isolated for just that behaviour instead of running in update?

Reply:
Understood. Unfortunate there isn't a callback, but oh well.

Reply:
I'll pass on that feedback to the tools team, see if they think it could be something they add in the future